gui/create-item.lua

```diff
@@ -184,11 +211,183 @@ local function getCreatureRaceAndCaste(caste)
 end
 
+function createCorpsePiece(creator, bodypart, partlayer, creatureID, casteID, generic, quality) -- this part was written by four rabbits in a trenchcoat (ppaawwll)
+ -- (partlayer is also used to determine the material if we're spawning a "generic" body part (i'm just lazy lol))
+ quality = math.max(0, math.min(5, quality - 1))
+ local creat = tonumber(creatureID)
+ -- get the actual raws of the target creature
+ local creatorRaceRaw = df.global.world.raws.creatures.all[creat]
+ local cast = tonumber(casteID)
+ local bodpart = tonumber(bodypart)
+ local patlayer = tonumber(partlayer)
+ -- get body info for easy reference
+ local creatorBody = creatorRaceRaw.caste[cast].body_info
+ 
+ local layerName
+ local layerMat
+ 
+ if not generic then -- if we have a specified body part and layer, figure all the stuff out about that
+ -- store the tissue id of the specific layer we selected
+  local parlayer = tonumber(creatorBody.body_parts[bodpart].layers[patlayer].tissue_id)
+  print(creatorRaceRaw.tissue[parlayer].id)
+  layerMat = creatorRaceRaw.tissue[parlayer].tissue_material_str[1]
+  layerName = creatorBody.body_parts[bodpart].layers[patlayer].layer_name
+ else -- otherwise, figure out the mat name from the dual-use partlayer argument
+  layerMat = creatorRaceRaw.material[patlayer].id
+  layerName = layerMat
+ end
+
+ -- default is MEAT, so if anything else fails to change it to something else, we know that the body layer is a meat item
+ local itemm = "MEAT"
+ -- get race name and layer name, both for finding the item material, and the latter for determining the corpsepiece flags to set
+ local raceName = string.upper(creatorRaceRaw.creature_id)
+
+ print(creatorRaceRaw.creature_id)
+
+ -- every key is a valid non-hair corpsepiece, so if we try to index a key that's not on the table, we don't have a non-hair corpsepiece
+ local corpseTable = {BONE = "BONE", SKIN = "SKIN", CARTILAGE = "CARTILAGE", TOOTH = "TOOTH", NERVE = "NERVE", NAIL = "NAIL", HORN = "HORN", HOOF = "HOOF"}
+ -- we do the same as above but with hair
+ local hairTable = {HAIR = "HAIR", EYEBROW = "EYEBROW", EYELASH = "EYELASH", MOUSTACHE = "MOUSTACHE", CHIN_WHISKERS = "CHIN_WHISKERS", SIDEBURNS = "SIDEBURNS"}
+ -- if the layer is fat, spawn a glob of fat and DON'T check for other layer types
+ if layerName == "FAT" then
+  itemm = "GLOB"
+ elseif corpseTable[layerName] or hairTable[layerName] then -- check if hair
+  -- farlayer = 5
+  -- layerName = "HAIR" -- we then simplify every hair-tissue into just "HAIR"
+  itemm = "CORPSEPIECE"
+ end
+ 
+ local itemType = dfhack.items.findType(itemm..":NONE")
+ local itemSubtype = dfhack.items.findSubtype(itemm..":NONE")
+ 
+ print(layerMat)
+ 
+ local material = "CREATURE_MAT:"..raceName..":"..layerMat
+ local materialInfo = dfhack.matinfo.find(material)
+ 
+ local item1 = dfhack.items.createItem(itemType, itemSubtype, materialInfo['type'], materialInfo.index, creator)
```

Comment:
Maybe `item_id`?

Reply:
it was like that in the original modtools/create-item, but true